File: FoodDlvAPI/Models/Services/CartService.cs
using FoodDlvAPI.Interfaces;
using FoodDlvAPI.Models;
using FoodDlvAPI.Models.DTOs;
using FoodDlvAPI.Models.ViewModels;
using Microsoft.AspNetCore.Mvc;
using NuGet.Protocol.Core.Types;

namespace FoodDlvAPI.Models.Services
{
    public class CartService
    {
        //Fields       
        private readonly ICartRepository _cartRepository;
        private readonly IProductRepository _productRepository;


        //Constructors
        public CartService(ICartRepository cartRepository,
                            IProductRepository productRepository)
        {
            _cartRepository = cartRepository;
            _productRepository = productRepository;
        }

        public void AddToCart(CartDTO request)
        {
            var cart = Current(request.MemberId, request.StoreId);
            _cartRepository.AddDetail(cart, request);
        }

        public List<CartDTO> CartInfos(int memberId)
        {            
            var cartInfo = _cartRepository.GetCartInfos(memberId);
            return cartInfo;
        }


        public CartDTO Current(int memberId, int storeId)
        {
            if (_cartRepository.IsExists(memberId, storeId))
            {
                return _cartRepository.Load(memberId, storeId);
            }
            else
            {
                return _cartRepository.CreateNewCart(memberId, storeId);
            }
        }

        public void UpdateCart(CartDTO request)
        {
            _cartRepository.RemoveDetail(request.Details.First().IdentifyNum);
            if (request.Details.First().Qty >= 1)
            {
                AddToCart(request);
            }
        }

        public void RemoveDetail(int identifyNum)
        {
            _cartRepository.RemoveDetail(identifyNum);
        }

        public void DeleteCart(int memberId, int storeId)
        {
            _cartRepository.EmptyCart(memberId, storeId);
        }

        public void CheckOutCart(int memberId, int storeId)
        {
            var cart = Current(memberId, storeId);
            if (cart.Details.Count == 0 || cart.Details == null)
            {
                throw new Exception("購物車內無商品, 無法進行結帳");
            }
        }


    }
}

File: food-dlv-website/src/services/Cart/cart.service.js
import axios from "axios";

const API_URL = "https://localhost:7093/api/Cart";

class CartService {
  postAddToCart(memberId, storeId, productId, itemId, qty) {
    const response = axios.post(API_URL + "/AddToCart", {
      MemberId: memberId,
      StoreId: storeId,
      ProductId: productId,
      ItemsId: itemId,
      Qty: qty,
    });
    return response;
  }

  getCartInfo(memberId, storeId) {
    const response = axios.get(
      API_URL + `/CartInfo?memberId=${memberId}&storeId=${storeId}`
    );
    return response;
  }

  postUpdateCart(identifyNum, storeId, productId, itemId, qty) {
    const response = axios.post(API_URL + "/UpdateCart", {
      IdentifyNum: identifyNum,
      StordId: storeId,
      ProductId: productId,
      ItemsId: itemId,
      Qty: qty,
    });
    return response;
  }

  async postRemoveDetail(identifyNum) {
    const response = await axios.post(
      API_URL + "/RemoveDetail" + `?identifyNum=${identifyNum}`
    );
    return response;
  }

  //   async deleteDeleteCart(memberId, storeId) {
  //     console.log(memberId);
  //     console.log(storeId);
  //     const response = await axios.delete(API_URL + "/DeleteCart", {
  //       memberId,
  //       storeId,
  //     });
  //     return response;
  //   }
  // }

  async deleteDeleteCart(memberId, storeId) {
    const response = await axios.delete(
      API_URL + "/DeleteCart" + `?memberId=${memberId}&storeId=${storeId}`
    );
    return response;
  }
}

export default new CartService();
